ecommerce/factories.py

```diff
@@ -87,3 +89,41 @@ class CoursePriceFactory(DjangoModelFactory):
 
     class Meta:  # pylint: disable=missing-docstring,no-init,too-few-public-methods,old-style-class
         model = CoursePrice
+
+
+class CouponFactory(DjangoModelFactory):
+    """Factory for Coupon"""
+    amount_type = FuzzyChoice(Coupon.AMOUNT_TYPES)
+    coupon_code = FuzzyChoice([None, FuzzyText().fuzz()])
+    num_redemptions_per_user = FuzzyInteger(1, 10)
+    num_coupons_available = FuzzyInteger(1, 100)
+
+    class Meta:  # pylint: disable=missing-docstring,no-init,too-few-public-methods,old-style-class
+        model = Coupon
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
```

Comment:
I removed the `content_type` and `object_id` pieces, these are automatically handled by the `GenericForeignKey`